mt-kahypar/partition/coarsening/multilevel_coarsener.h

```diff
@@ -374,32 +374,34 @@ class MultilevelCoarsener : public ICoarsener,
     HypernodeWeight weight_v = _cluster_weight[v];
     if ( weight_u + weight_v <= _context.coarsening.max_allowed_node_weight ) {
 
-      if ( _matching_state[u].compare_exchange_strong(unmatched, match_in_progress) ) {
+      uint8_t expect_unmatched_u = STATE(MatchingState::UNMATCHED);
+      if ( _matching_state[u].compare_exchange_strong(expect_unmatched_u, match_in_progress) ) {
         _matching_partner[u] = v;
         // Current thread gets "ownership" for vertex u. Only threads with "ownership"
         // can change the cluster id of a vertex.
 
+        uint8_t expect_unmatched_v = STATE(MatchingState::UNMATCHED);
         uint8_t matching_state_v = _matching_state[v].load();
-        if ( matching_state_v == STATE(MatchingState::MATCHED) ) {
+        if ( matching_state_v == matched ) {
           // Vertex v is already matched and will not change it cluster id any more.
           // In that case, it is safe to set the cluster id of u to the cluster id of v.
           const HypernodeID rep = cluster_ids[v];
-          ASSERT(_matching_state[rep] == STATE(MatchingState::MATCHED));
+          ASSERT(_matching_state[rep] == matched);
           success = joinCluster<has_fixed_vertices>(hypergraph,
             u, rep, cluster_ids, contracted_nodes, fixed_vertices);
-        } else if ( _matching_state[v].compare_exchange_strong(unmatched, match_in_progress) ) {
+        } else if ( _matching_state[v].compare_exchange_strong(expect_unmatched_v, match_in_progress) ) {
```

Comment:
Specify a memory ordering here. Acquire should be right here.

Reply:
I added memory orderings for the whole file. For this specific line even relaxed suffices, if I'm not mistaken